pkg/cloudprovider/aws/instance.go

```diff
@@ -185,10 +254,14 @@ func (p *InstanceProvider) getOverrides(instanceTypeOptions []cloudprovider.Inst
 				continue
 			}
 			for _, subnet := range subnets {
-				if aws.StringValue(subnet.AvailabilityZone) == offering.Zone {
+				subnetZone := aws.StringValue(subnet.AvailabilityZone)
+				if subnetZone == offering.Zone && constrainedZones.Has(subnetZone) {
```

Comment:
Thoughts on shortcircuiting zonal matches similar to how we do w/ capacity type on 253

```
if offering.CapacityType != capacityType {
  continue
}
if !constrainedZones.Has(offering.Zone) {
  continue
}
for _, subnet := range subnets {
  if  offering.Zone == aws.StringValue(subnet.AvailabilityZone) {
    override = ...
  }
}
```